src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import javax.naming.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtils;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final int PASSWORD_MIN_LENGTH = 6;
+    private static final String REGEX_PATTERN = "^(.+)@(\\w+)$";
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> user = userService.findByEmail(email);
+        user.orElseThrow(AuthenticationException::new);
+        String hashedPassword = HashUtils.hashPassword(password, user.get().getSalt());
+        if (!user.get().getPassword().equals(hashedPassword)) {
+            throw new AuthenticationException("Invalid login or password");
+        } else {
+            return user.get();
+        }
```

Comment:
can be simplified 
in the 23 line you're throwing an exception in case user is empty, but you don't assign the value if it exists. You could have saved the variable with the user if it exists and avoid using user.get()